elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -149,35 +228,78 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
+            final String credentialStore = CREDENTIAL_STORE.resolveModelAttribute(context, model).asStringOrNull();
+            final String secretKey = SECRET_KEY.resolveModelAttribute(context, model).asStringOrNull();
+            final String keyStoreName = KEY_STORE.resolveModelAttribute(context, model).asStringOrNull();
+            final String keyPairAlias = KEY_STORE_ALIAS.resolveModelAttribute(context, model).asStringOrNull();
 
+            final InjectedValue<KeyStore> keyStoreInjector = new InjectedValue<>();
             final InjectedValue<PathManager> pathManagerInjector = new InjectedValue<>();
             final InjectedValue<NameRewriter> nameRewriterInjector = new InjectedValue<>();
 
+            SecretKey key = null;
+            if (credentialStore != null && secretKey != null) {
+                key = getSecretKey(context, credentialStore, secretKey);
+            }
+            SecretKey finalKey = key;
+            ServiceRegistry keyStoreServiceRegistry = context.getServiceRegistry(true);
+
+
             TrivialService<SecurityRealm> fileSystemRealmService = new TrivialService<>(
                     new TrivialService.ValueSupplier<SecurityRealm>() {
-
                         private PathResolver pathResolver;
-
+                        ModifiableKeyStoreService keyStoreService;
                         @Override
                         public SecurityRealm get() throws StartException {
+
                             pathResolver = pathResolver();
                             Path rootPath = pathResolver.path(path).relativeTo(relativeTo, pathManagerInjector.getOptionalValue()).resolve().toPath();
 
                             NameRewriter nameRewriter = nameRewriterInjector.getOptionalValue();
                             Charset charset = Charset.forName(hashCharset);
                             Encoding encoding = HEX.equals(hashEncoding) ? Encoding.HEX : Encoding.BASE64;
+                            if (nameRewriter == null) {
+                                nameRewriter = NameRewriter.IDENTITY_REWRITER;
+                            }
+                            KeyStore keyStore = keyStoreInjector.getOptionalValue();
+                            PrivateKey privateKey = null;
+                            PublicKey publicKey = null;
+                            if (keyStore != null) {
+                                    try {
+                                        keyStoreService = CertificateAuthorityAccountDefinition.getModifiableKeyStoreService(keyStoreServiceRegistry, keyStoreName);
+                                        char[] keyPassword = getKeyStorePassword((KeyStoreService) keyStoreService);
+                                        privateKey = (PrivateKey) keyStore.getKey(keyPairAlias, keyPassword);
+                                        publicKey = keyStore.getCertificate(keyPairAlias).getPublicKey();
+                                        if (privateKey == null ^ publicKey == null) {
```

Comment:
Is this check necessary if I have the catch statement below on line 281?

Reply:
Both getKey and getCertificate can return null if the alias and certificate aren't found. So we do need to validate these since an exception wouldn't be thrown in this case.